### ISSUE TITLE
Implement `open`

### ISSUE THREAD

**Issue by certik** [prefix-dev/shell#109]
On macOS the `open` command uses the system-registered GUI program to open a given file. On Windows in WSL one can use `explorer some.xls`. On Linux there is also a way to do this.

So we can implement this support on all platforms and just call it `open`.

**Comment by certik** [comment]
The `open` command could be a conda package. Then we are in the subset category again. :)

Alternatively we can only support it in interactive mode, I suspect I never used "open" in a script on macOS, only in interactive mode.

**Comment by wolfv** [comment]
On Linux it's usually `xdg-open` ... which is quite unintuitive.

**Comment by wolfv** [comment]
Ok, I have some more thoughts on this now.

Maybe we should create a "coreutils-ext" that contains some additional utilities such as:

- `dl`: download a file to stdio
- `open`: open a file (`open`/`xdg-open`/`explorer.exe` replacement)
- ...

That way, they could also be installed separately by users of bash/zsh and don't hinder compatibility concerns.

**Comment by prsabahrami** [comment]
> Ok, I have some more thoughts on this now.
> 
> 
> 
> Maybe we should create a "coreutils-ext" that contains some additional utilities such as:
> 
> 
> 
> - `dl`: download a file to stdio
> 
> - `open`: open a file (`open`/`xdg-open`/`explorer.exe` replacement)
> 
> - ...
> 
> 
> 
> That way, they could also be installed separately by users of bash/zsh and don't hinder compatibility concerns.

I like this =))

**Comment by certik** [comment]
Yes, that's the way to do it. Anything we do must work in regular Bash and Zsh on Linux / macOS. Most of these things can be implemented as standalone programs.

**Comment by certik** [comment]
> Maybe we should create a "coreutils-ext" that contains some additional utilities such as:

I made this https://github.com/prefix-dev/shell/issues/141.
